views/pages/EditConfigPage.jsx

```diff
@@ -216,340 +227,350 @@ class EditConfig extends Component {
+    window.addEventListener('resize', this.handleResize)
   }
   componentWillUnmount() {
-    window.removeEventListener('resize', this.handleResize);
+    window.removeEventListener('resize', this.handleResize)
   }
   handleResize = () => {
     if (window.innerWidth >= 768) {
-      let gridCols = Math.floor(window.innerWidth / this.state.gridWidth);
+      let gridCols = Math.floor(window.innerWidth / this.state.gridWidth)
       this.setState({
-        gridCols: gridCols
-      });
+        gridCols: gridCols,
+      })
     } else {
       this.setState({
-        gridCols: 1
-      });
+        gridCols: 1,
+      })
     }
   }
   brewColors = () => {
-    let blocks = [];
-    let colors = [];
+    let blocks = []
+    let colors = []
     for (let scales in colorbrewer) {
       for (let degree in colorbrewer[scales]) {
         if (!isNaN(parseFloat(degree)) && !isNaN(degree - 0)) {
           blocks.push(
-            <MenuItem
-              value={blocks.length}
-            >
+            <MenuItem value={blocks.length}>
               <div
                 style={{
                   display: 'table',
-                  width: '100%'
+                  width: '100%',
                 }}
               >
                 {this.getBlocks(colorbrewer[scales][degree])}
               </div>
             </MenuItem>
-          );
-          colors.push(colorbrewer[scales][degree]);
+          )
+          colors.push(colorbrewer[scales][degree])
         }
       }
     }
     this.setState({
       colorblock: blocks,
-      colorarray: colors
-    });
+      colorarray: colors,
+    })
   }
   getBlocks = (colors) => {
-    let block = [];
+    let block = []
     for (let color in colors) {
       block.push(
         <span
           style={{
             width: '20px',
             height: '20px',
             display: 'table-cell',
-            backgroundColor: colors[color]
+            backgroundColor: colors[color],
           }}
-        >
-        </span>
-      );
+        ></span>
+      )
     }
-    return block;
+    return block
   }
   fetchConfigurations = (uid, config) => {
-    return window.fetch(`${basePath}/api/v1/users/${uid}/configs/${config}`, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      credentials: 'same-origin'
-    }).then((response) => {
-      if (response.status !== 200) {
-        return;
-      }
-      return response.json();
-    }).then((response) => {
-      response['readers'] = response['readers'].map(friend => ({
-        label: friend,
-        value: friend
-      }));
-      this.setState(response);
-      let configKey = this.getConfigKey(response['config']); //temp
-      this.setState({
-        themes: this.getDefaultTheme(response['config'][configKey]),
-        viewOnly: response['owner'] !== this.props.user.uid
-      });
-    });
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/configs/${config}`, {
```

Comment:
route path

Reply:
also can we use the model object for this now?

Reply:
I'll add that to the todo for when this page is refactored.